### ISSUE TITLE
Memory error for peek_char/1

### ISSUE THREAD

**Issue by dram** [trealla-lang/trealla#231]
Valgrind detect memory error for following code, although it run successfully:

```
:- initialization(main).

main :-
        see('foo.pro'),
        peek_char(C),
        write(C), nl,
        halt.
```

Valgrind output:

```
==27736== Invalid read of size 1
==27736==    at 0x1142F7: needs_quoting (parser.c:158)
==27736==    by 0x15373C: unify_atom (jela.c:593)
==27736==    by 0x12A129: bif_iso_peek_char (bifs_iso.c:2118)
==27736==    by 0x15321A: call (jela.c:473)
==27736==    by 0x1533F1: run_me (jela.c:525)
==27736==    by 0x1119F1: query_run (trealla.c:935)
==27736==    by 0x11302E: trealla_run_query (trealla.c:1425)
==27736==    by 0x112C74: trealla_consult_file (trealla.c:1329)
==27736==    by 0x10E41B: main (tpl.c:363)
==27736==  Address 0x6082360 is 0 bytes inside a block of size 2 free'd
==27736==    at 0x4C2CDDB: free (vg_replace_malloc.c:530)
==27736==    by 0x114965: make_atom (parser.c:327)
==27736==    by 0x15372C: unify_atom (jela.c:591)
==27736==    by 0x12A129: bif_iso_peek_char (bifs_iso.c:2118)
==27736==    by 0x15321A: call (jela.c:473)
==27736==    by 0x1533F1: run_me (jela.c:525)
==27736==    by 0x1119F1: query_run (trealla.c:935)
==27736==    by 0x11302E: trealla_run_query (trealla.c:1425)
==27736==    by 0x112C74: trealla_consult_file (trealla.c:1329)
==27736==    by 0x10E41B: main (tpl.c:363)
==27736==  Block was alloc'd at
==27736==    at 0x4C2BBAF: malloc (vg_replace_malloc.c:299)
==27736==    by 0x5AD8389: strdup (strdup.c:42)
==27736==    by 0x12A10C: bif_iso_peek_char (bifs_iso.c:2118)
==27736==    by 0x15321A: call (jela.c:473)
==27736==    by 0x1533F1: run_me (jela.c:525)
==27736==    by 0x1119F1: query_run (trealla.c:935)
==27736==    by 0x11302E: trealla_run_query (trealla.c:1425)
==27736==    by 0x112C74: trealla_consult_file (trealla.c:1329)
==27736==    by 0x10E41B: main (tpl.c:363)
==27736==
```

**Comment by infradig** [comment]
Not seeing this.

**Comment by dram** [comment]
I have checked that this issue is fixed with issue #230 by this commit https://github.com/trealla-lang/trealla/commit/d6203153d82a7199ea162728f4d59456d63b404f

So closing it.
